src/query/storages/parquet/src/parquet2/processors/deserialize_transform.rs

```diff
@@ -148,6 +149,39 @@ impl Parquet2DeserializeTransform {
         Ok(())
     }
 
+    fn process_small_groups(&mut self, part: &Parquet2SmallGroupPart) -> Result<Vec<DataBlock>> {
+        let mut blocks = Vec::new();
+        for (path, data) in part.groups.iter() {
+            blocks.extend(self.process_small_group(path.as_str(), data)?);
+        }
+        Ok(blocks)
+    }
+
+    fn process_small_group(&mut self, path: &str, groups: &Vec<usize>) -> Result<Vec<DataBlock>> {
+        let mut res = Vec::new();
+        let builder = Memory::default();
+        let op = Operator::new(builder)?.finish();
+        let blocking_op = op.blocking();
+        let parts = self.partition_pruner.prune_file_group(path, &op, groups)?;
+
+        for part in parts {
+            let mut readers = self
+                .source_reader
+                .row_group_readers_from_blocking_io(&part, &blocking_op)?;
+            if let Some(block) = self.process_row_group(&part, &mut readers)? {
+                res.push(block)
+            }
+        }
```

Comment:
Small groups should be read in one IO like small files. `row_group_readers_from_blocking_io` will conduct multiple IOs. cc  @youngsofun @sundy-li

Reply:
> Small groups should be read in one IO like small files
it is right for copy, all columns will be loaded.

for select, `merge io` as in fuse is more general.

Reply:
@zenus @RinChanNOWWW is right, these optimizations (small file, small rowgroups, column merge io) aim to reduce the number of IO.

Reply:
i see, i working  on read_columns_data_by_merge_io, the code would be pushed later.

Reply:
@youngsofun  completed, please help check.